### PR TITLE
953 dot path

### LINKED ISSUE #953: NPE bug if foldername contains a dot
### Expected behavior

CobiGen should work and not fail with NPE

### Actual behavior

If I have a template in a folder that has a dot in its name, I get an NPE:
```
java.lang.NullPointerException
  java.lang.NullPointerException
      at java.util.ArrayList.addAll(ArrayList.java:581)
      at com.devonfw.cobigen.eclipse.wizard.common.SelectFilesPage.getTemplatesToBeGenerated(SelectFilesPage.java:271)
      at com.devonfw.cobigen.eclipse.wizard.generate.GenerateWizard.generateContents(GenerateWizard.java:75)
      at com.devonfw.cobigen.eclipse.wizard.generate.common.AbstractGenerateWizard.performFinish(AbstractGenerateWizard.java:65)
      at org.eclipse.jface.wizard.WizardDialog.finishPressed(WizardDialog.java:775)
      at org.eclipse.jface.wizard.WizardDialog.buttonPressed(WizardDialog.java:414)
      at org.eclipse.jface.dialogs.Dialog.lambda$0(Dialog.java:622)
      at org.eclipse.swt.events.SelectionListener$1.widgetSelected(SelectionListener.java:84)
      at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:252)
      at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
      at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4292)
      at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1512)
      at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1535)
      at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1520)
      at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:1324)
      at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4116)
      at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3733)
      at org.eclipse.jface.window.Window.runEventLoop(Window.java:821)
      at org.eclipse.jface.window.Window.open(Window.java:797)
      at com.devonfw.cobigen.eclipse.workbenchcontrol.handler.GenerateHandler.execute(GenerateHandler.java:84)
      at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:294)
      at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:96)
      at sun.reflect.GeneratedMethodAccessor296.invoke(Unknown Source)
      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
      at java.lang.reflect.Method.invoke(Method.java:498)
      at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
      at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:320)
      at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:254)
      at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:164)
      at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:156)
      at org.eclipse.core.commands.Command.executeWithChecks(Command.java:497)
      at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:490)
      at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:207)
      at org.eclipse.e4.ui.workbench.renderers.swt.HandledContributionItem.executeItem(HandledContributionItem.java:436)
      at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:449)
      at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.lambda$2(AbstractContributionItem.java:475)
      at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
      at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4292)
      at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1512)
      at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1535)
      at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1520)
      at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:1324)
      at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4116)
      at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3733)
      at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1173)
      at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
      at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1062)
      at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:156)
      at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:628)
      at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
      at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:563)
      at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:151)
      at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:155)
      at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:199)
      at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:137)
      at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:107)
      at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:391)
      at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:246)
      at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
      at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
      at java.lang.reflect.Method.invoke(Method.java:498)
      at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
      at org.eclipse.equinox.launcher.Main.basicRun(Main.java:595)
      at org.eclipse.equinox.launcher.Main.run(Main.java:1501)
```

### Steps to reproduce (bug) / Use Case of feature request (enhancement)

1. I created a template to generate a flyway SQL
2. As I did not know the version I put it into: resources/db/migration/x.y/V00xx__CreateTable_${variables.entityName}.sql.ftl 
3. I added it to my templates.xml
4. I ran CobiGen
5. Already the preview in Eclipse seems broken: It shows the generation until the folder "x.y" but not the template inside just as if the folder itself would be the template.
6. When I click generate, I get the NPE above.

### Related/Dependent Issues

none.

### PR DESCRIPTION
**Pull request by LarsReinken**
Adresses/Fixes #953.

Implements

* 
* 
* 

@devonfw/cobigen

### CODE REVIEW

File: cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/wizard/common/model/SelectFileContentProvider.java [hunk 1]
```diff
@@ -461,9 +462,14 @@ public Object getProvidedObject(String path) {
      * @return <code>true</code> if the last element of the path contains a dot<br>
      *         <code>false</code>, otherwise
      */
-    private boolean targetIsFile(IPath path) {
-
-        return path.lastSegment().contains(".");
+    private boolean targetIsFile(IPath path, List<String> paths) {
+        int index = paths.indexOf(path.toString());
+        if (index + 1 <= paths.size() - 1) {
```
Comment (jdiazgon) [inline cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/wizard/common/model/SelectFileContentProvider.java]: Can you explain me the logic behind that `if`?

Wouldn't be the same with the following?:
`if (index < path.size())`

File: cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/wizard/common/model/SelectFileContentProvider.java [hunk 2]
```diff
@@ -461,9 +462,14 @@ public Object getProvidedObject(String path) {
      * @return <code>true</code> if the last element of the path contains a dot<br>
```
Comment (jdiazgon) [inline cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/wizard/common/model/SelectFileContentProvider.java]: Remember to update the Javadoc accordingly with the changes. You have added a new parameter and changed the logic.

File: cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/wizard/common/model/SelectFileContentProvider.java (hunk 1 above)
Comment (maybeec) [reply cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/wizard/common/model/SelectFileContentProvider.java]: I think the only correct subsitution would be if (index < paths.size() -1).
Your's @jdiazgon would be failing.

Comment (jdiazgon) [reply cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/wizard/common/model/SelectFileContentProvider.java]: Ooh yeah, you are right

**Pull request merged**